manifests/init.pp

```diff
@@ -155,6 +158,7 @@
   $user_hash          = {},
   $configure_firewall = undef,
   $install_java       = $jenkins::params::install_java,
+  $java_package_name  = undef,
```

Comment:
You'd probably want to make it only default to 1.7 on cent/rhel 5.    most of the rest use default, and 1.7 is deprecated arlready.  newer ubuntu's use 1.8, for example.  

But that's what params is for, doing OS based defaults :)